addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java

```diff
@@ -73,18 +75,27 @@ public static int getPortForSite(String site) {
         if (site == null) {
             return 80;
         }
-        String[] schemeHostPort = site.split(":");
-        if (schemeHostPort.length == 3) {
-            try {
-                return Integer.parseInt(schemeHostPort[2]);
-            } catch (NumberFormatException e) {
-                // Ignore
+
+        try {
+            var url = new URL(site);
+            var port = url.getPort();
+            if (port != -1) {
+                return port;
+            }
+
+            if (url.getProtocol().equals("https")) {
```

Comment:
This should probably use an ignorecase variant, which may mean using StringUtils